Notebook: publish draft

tilen1976 opened the pull request:
This notebook demonstrates two methods for publishing a variable definition from draft
- Code based method
- File based method

Code based method is definitely the easiest (not so many possibilities for errors) when only updating one field so this chapter is placed before file based method.

Allow some duplication for minimizing user errors:
- getting variable definition for each method

Use variables to support user, but try to avoid errors caused by running cells by accident for example like this:

![Screenshot 2025-03-07 at 10 57 29](https://github.com/user-attachments/assets/c636b132-70f9-4634-ac9d-9cb8c3b26e7e)


Add index at top with links:
- doesn't always work, but it should in theory. 
- Errors probably caused by caching/metadata saved in notebooks
- Are still useful for navigation (are highlighted anyway)

No custom directory for saving file since it is not implemented yet for method

mmwinther commented:
I really like the introduction, very useful information! We should use that format across all our notebooks.

A few comments:

- I don't think we should have a manually updated `Innhold` section because it can quickly get outdated. Both VSCode and Jupyter offer ways of viewing the structure of a Notebook based on the Markdown headings.
- It looks like you're going for a pedagogical style where we show the user the options available to them and let them choose. But I think we're giving the user too many options and that it can be overwhelming. We should think of these notebooks as tools that the user will use to get the job done. With that in mind I think we should:
  - Only get a variable by `short_name`
  - Have one code block for Publisere internt and one for Publisere eksternt
  - Remove the oppdatering block (there is both a confirmation log and the whole `VariableDefinition` object is returned from the `update_draft` method)

While we're here, I think it would be relatively simple for us and very helpful for the user if we implemented methods on the `VariableDefinition` class: `publish_external()` and `publish_internal()`, what do you think?

tilen1976 commented:
> I really like the introduction, very useful information! We should use that format across all our notebooks.
> 
> A few comments:
> 
> * I don't think we should have a manually updated `Innhold` section because it can quickly get outdated. Both VSCode and Jupyter offer ways of viewing the structure of a Notebook based on the Markdown headings.
> * It looks like you're going for a pedagogical style where we show the user the options available to them and let them choose. But I think we're giving the user too many options and that it can be overwhelming. We should think of these notebooks as tools that the user will use to get the job done. With that in mind I think we should:
>   
>   * Only get a variable by `short_name`
>   * Have one code block for Publisere internt and one for Publisere eksternt
>   * Remove the oppdatering block (there is both a confirmation log and the whole `VariableDefinition` object is returned from the `update_draft` method)
>   * Remove the Fil-basert metode section, it seems like more trouble than it's worth and we should recommend one good way of doing things
> 
> While we're here, I think it would be relatively simple for us and very helpful for the user if we implemented methods on the `VariableDefinition` class: `publish_external()` and `publish_internal()`, what do you think?

I have updated the changes based on most of the comments, but I would like to keep the last block (at least for now), where the user can confirm the new status. The response is not very clear, and after removing most of the confusing options, this block doesn’t add much overhead or take up much space.

mmwinther reviewed (approved):
Looks awesome, nice one 🙌
